addons/mail/static/src/core/common/record.js

```diff
@@ -435,31 +392,66 @@ export class RecordList extends Array {
 }
 
 /**
- * @typedef {Object} RelationDefinition
- * @property {string} [targetModel]
- * @property {Function} [compute]
- * @property {string} [inverse]
- * @property {Function} [onAdd]
- * @property {Function} [onDelete]
+ * @typedef {Object} FieldDefinition
+ * @property {boolean} [ATTR_SYM] true when this is an attribute, i.e. a non-relational field.
+ * @property {boolean} [MANY_SYM] true when this is a many relation.
+ * @property {boolean} [ONE_SYM] true when this is a one relation.
+ * @property {any} [default] the default value of this attribute.
+ * @property {any} [html] whether the attribute is an html field. Useful to automatically markup
```

Comment:
`{boolean}` (and maybe undefined?)

Reply:
`{boolean}` feels enough, with `[]` around name acting as optional, and when no explicit one is set we assume it's undefined